File: NapakalakiJava/src/NapakalakiGame/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author danibolanos & jomabose
 */
public class CultistPlayer extends Player{
    private static int totalCultistPlayers=0;
    Cultist myCultistCard;
    
    public CultistPlayer(Player p, Cultist c){
       super(p);
       myCultistCard = c;
       totalCultistPlayers++;
    }
    @Override
    public int getCombatLevel(){
        int levelPlayerAdded = (int) (super.getCombatLevel()*0.7);
        int combatLevel = super.getCombatLevel() + levelPlayerAdded + 
        myCultistCard.getGainedLevels()*totalCultistPlayers;
        
        return combatLevel;
    }
    @Override
    protected int getOponentLevel(Monster m){
        return m.getCombatLevelAgainstCultistPlayer();
    }
    @Override
    protected boolean shouldConvert(){
        return false;
    }
    @Override
    protected Treasure giveMeATreasure(){
        int indice=(int) (Math.random()*getVisibleTreasures().size()-1);
        Treasure tesoro = getVisibleTreasures().get(indice);
        getVisibleTreasures().remove(indice);
        
        return tesoro;        
    }
    @Override
    protected boolean canYouGiveMeATreasure(){
        boolean puedo=true;
        
        if(getVisibleTreasures().isEmpty())
            puedo=false;
        
        return puedo;       
    }
    public static int getTotalCultistPlayers(){
        return totalCultistPlayers;
    }
    @Override
    public String toString(){
        String cadena = super.toString();
        cadena += "\n\n" + "*" + this.getName()+" es Sectario*";
        
        return cadena;
    }
}

File: EXAMENES_PDOO/NapakalakiRuby/lib/cultist_player.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
  
require_relative 'player'
require_relative 'cultist'

  class CultistPlayer < Player
    @@totalCultistPlayers = 0
    
    def initialize(p,c)
      @name = p.getName
      @level = p.getLevels
      @dead = p.isDead 
      @pendingBadConsequence = p.pendingBadConsequence
      @visibleTreasures = p.getVisibleTreasures
      @hiddenTreasures = p.getHiddenTreasures
      @canISteal = p.canISteal
      @enemy = p.enemy
      @myCultistCard = c
      @@totalCultistPlayers += 1
    end
    
    def getTotalCultistPlayers
      @@totalCultistPlayers
    end
    protected
    
    def getCombatLevel
      cl = (super * 1.7).truncate
      cl += @@totalCultistPlayers * @myCultistCard.getGainedLevels
    end
    
    def getOponentLevel(m)
      m.getCombatLevelAgainstCultistPlayer
    end
    
    def shouldConvert
      false
    end
    
    private
    
    def giveMeATreasure
      indice = rand(@visibleTreasures.length)
      t = @visibleTreasures.at(indice)
      discardVisibleTreasure(t)
      t
    end
    
    def canYouGiveMeATreasure
      puedo=true
      if @visibleTreasures.empty? then
        puedo=false
      end
      puedo
    end
  end
end
